### PR TITLE
Convert execute_resource remaining properties to use properties 

### LINKED ISSUE #7063: Convert execute resource to setup properties via property DSL
The execute resource has not been fully converted to using property to set properties. We should convert this to setup properties via the property resource DSL.

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@msystechnologies.com>

### Description
The execute resource has not been fully converted to using property to set properties. We should convert this to setup properties via the property resource DSL.

### Issues Resolved
Fixes #7063
### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/chef/resource/execute.rb [hunk 1]
```diff
@@ -23,10 +23,6 @@ class Chef
   class Resource
     class Execute < Chef::Resource
       resource_name :execute
-      provides :execute
```
Comment (tas50) [inline lib/chef/resource/execute.rb]: ?

Comment (Vasu1105) [reply lib/chef/resource/execute.rb]: I didn't felt removing this is making any difference and it's required as I compared this with other resources. Let me know if I am missing something here.

Comment (lamont-granquist) [reply lib/chef/resource/execute.rb]: it is technically redundant with resource_name, but I'd like to deprecate resource_name, so having the `provides :execute` line in here is more forwards-looking.

Comment (Vasu1105) [reply lib/chef/resource/execute.rb]: @lamont-granquist Fixed.

**Pull request merged**